packages/nimble-components/src/table/index.ts

```diff
@@ -171,6 +172,12 @@ export class Table<
     @observable
     public firstSortedColumn?: TableColumn;
 
+    /**
+     * @internal
+     */
+    @observable
+    public documentShiftKeyDown = false;
```

Comment:
Just a thought scanning the draft PR, I think the state associated with the current keyboard interactions could keep growing in complexity. Certain keys do certain things based on row vs group header vs heirarchy row, vs column selected vs row selected, etc.

Should we use a state machine to model the current interaction state?
The `@xstate/fsm` packages is a good lightweight one with good TypeScript support https://xstate.js.org/docs/packages/xstate-fsm/#features 

We could opt-in to the full xstate package if any of those features would be helpful but I expect it might be overkill for this use-case.

Reply:
Woa.... there is a headless ui state machine library for modeling the aria interaction states of components and seems to be built on xstate/fsm 🤯

https://github.com/chakra-ui/zag/tree/main/packages/core

Reply:
After looking through the libraries you referenced, my current thought is that we shouldn't try to introduce a state machine in the table. Some of my thoughts are:
- It would be a very different approach to eventing/interactions than exists today in nimble, so I think unless there is a very compelling reason to introduce this new approach, I would lean towards consistency because I think that is what will make our code the most understandable and approachable (and therefore maintainable).
- We have the benefit of having subcomponents to help us manage our states. Therefore, I'm not sure we want to try to create a giant state machine when we already have a fairly clear way to break up all the different interactions that can happen in different parts of the component. For example, a `<nimble-group-row>` can encapsulate all its interactions, which can be different than all the interactions encapsulated within a `<nimble-row>`.

We can sync up offline if you want to discuss this some more. But, if we decide that a state machine library does make sense, then I'd push for introducing it in a separate PR. I think the introduction of a state machine library would be a large enough change that it would be better as a PR that isn't also introducing a bunch of new logic (like this one is).

Reply:
Yea we can discuss it offline (likely tomorrow). I don't think we should use a state machine to track the entire state of the table (just thought those component library was interesting, but it tries to manage the entire state which would be a huge change).

I was thinking we use a finite state machine just to encapsulate the state of the keyboard interactions and actions associated with what currently has tab focus. I think it may be easier to track the behavior of key interactions from one spot, particularly if we consider additional interaction modes in the future, i.e. grid vs table and maybe even extensions for touch interactions, etc.